core/state/statedb.go

```diff
@@ -1220,6 +1248,9 @@ func (s *StateDB) GetTrie() Trie {
 // The associated block number of the state transition is also provided
 // for more chain context.
 func (s *StateDB) Commit(block uint64, deleteEmptyObjects bool) (common.Hash, error) {
+	if s.arbTxFilter == blockFiltered {
```

Comment:
what about txFiltered?
I think this should also return an error as we expect filtered transactions to be reverted using RevertToSnapshot

Reply:
txFiltered shouldn't be hit if managed correctly (i.e revert snapshot is called promptly after a failure), but its safer to error out anyway.